File: app/src/main/java/alex/com/githubchecker/components/github/dagger/GithubModule.java
package alex.com.githubchecker.components.github.dagger;

import alex.com.githubchecker.components.app.api.APIClient;
import alex.com.githubchecker.components.github.repo.GithubModel;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Alex on 11/11/2017.
 */

@Module
public class GithubModule {

    @Provides
    @GithubScope
    GithubModel provideGithubModel(APIClient apiClient) {
        return new GithubModel(apiClient);
    }

}

File: app/src/main/java/alex/com/githubchecker/components/github/dagger/GithubModule.kt
package alex.com.githubchecker.components.github.dagger

import alex.com.githubchecker.components.app.api.APIClient
import alex.com.githubchecker.components.app.data.SessionDatamanager
import alex.com.githubchecker.models.dagger.GithubModel
import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides

/**
 * Created by Alex on 11/11/2017.
 */

@Module
class GithubModule {

    @Provides
    @GithubScope
    fun provideGithubModel(application: Application, apiClient: APIClient, sessionDatamanager: SessionDatamanager): GithubModel {
        return GithubModel(application, apiClient, sessionDatamanager)
    }
}
